sycl/source/CMakeLists.txt

```diff
@@ -80,6 +81,7 @@ set(SYCL_SOURCES
     "detail/usm/usm_dispatch.cpp"
     "detail/usm/usm_impl.cpp"
     "detail/util.cpp"
+    "cg.cpp"
```

Comment:
The corresponding header file resides in `CL/sycl/detail/`. Have you considered putting `cg.cpp` to `source/detail`?